R/get_event.R

```diff
@@ -49,38 +52,45 @@ get_event <- function(event_type='port_visit',
   # Extract all entries from list of responses
   all_entries <- purrr::map(all_results, purrr::pluck, 'entries') %>% purrr::flatten(.)
 
-  # Create progress bar
-  pb <- progress::progress_bar$new(
-    format = "Processing events: [:bar] :current/:total (:percent)",
-    total = length(all_entries)
-    )
+  # Process results if they exist
+  if(length(all_entries) > 0){
 
-  # Function to extract each entry to tibble
-  event_entry <- function(x){
+    # Convert list to dataframe
     df_out <- tibble::tibble(
-      id = x$id,
-      type = x$type,
-      start = x$start,
-      end = x$end,
-      lat = x$position$lat,
-      lon = x$position$lon,
-      regions = list(x$regions),
-      boundingBox = list(x$boundingBox),
-      distances = list(x$distances),
-      vessel = list(x$vessel),
-      event_info = list(x[length(x)])
+      id = purrr::map_chr(all_entries, 'id'),
```

Comment:
This works well... another option I think works is `purrr::map_chr(all_entries, purrr::pluck, 'id')`. Your's is simpler here so probably leave it.. but below its possible to use pluck rather than a nested map